src/Sulu/Bundle/MediaBundle/Media/ImageConverter/MediaImageExtractor.php

```diff
@@ -104,77 +95,82 @@ private function convertPdfToImage($content)
             );
         }
 
-        return $output;
+        return $this->createTemporaryResource($output);
     }
 
     /**
      * Converts a PSD to a png using imagine. Only works with Imagick and not with GD.
      *
-     * @param string $content
+     * @param resource $resource
      *
-     * @return string
+     * @return resource
      *
      * @throws InvalidMimeTypeForPreviewException
      */
-    private function convertPsdToImage($content)
+    private function convertPsdToImage($resource)
     {
-        $temporaryFilePath = $this->createTemporaryFile($content);
+        $temporaryFilePath = $this->createTemporaryFile($resource);
 
         try {
             $image = $this->imagine->open($temporaryFilePath);
             $image = $image->layers()[0];
 
             unlink($temporaryFilePath);
 
-            return $image->get('png');
+            return $this->createTemporaryResource($image->get('png'));
         } catch (RuntimeException $e) {
             unlink($temporaryFilePath);
 
             throw new InvalidMimeTypeForPreviewException('image/vnd.adobe.photoshop');
         }
     }
 
-    private function convertSvgToImage($content)
+    /**
+     * Converts one frame of a video to an image using FFMPEG.
+     *
+     * @param resource $resource
+     *
+     * @return resource
+     */
+    private function convertVideoToImage($resource)
     {
-        $temporaryFilePath = $this->createTemporaryFile($content);
+        $temporaryFilePath = $this->createTemporaryFile($resource);
+        $this->videoThumbnail->generate($temporaryFilePath, '00:00:02:01', $temporaryFilePath);
 
-        $image = $this->imagine->open($temporaryFilePath);
+        $extractedImage = file_get_contents($temporaryFilePath);
         unlink($temporaryFilePath);
 
-        return $image->get('png');
+        return $this->createTemporaryResource($extractedImage);
     }
 
     /**
-     * Converts one frame of a video to an image using FFMPEG.
+     * Create temporary resource which will removed on fclose or end of process.
      *
      * @param string $content
      *
-     * @return string
-     *
-     * @throws OriginalFileNotFoundException
+     * @return resource
      */
-    private function convertVideoToImage($content)
+    private function createTemporaryResource($content)
     {
-        $temporaryFilePath = $this->createTemporaryFile($content);
-        $this->videoThumbnail->generate($temporaryFilePath, '00:00:02:01', $temporaryFilePath);
-
-        $extractedImage = file_get_contents($temporaryFilePath);
-        unlink($temporaryFilePath);
+        $tempResource = tmpfile();
```

Comment:
Why do we need a temporary resource now?

Reply:
at some cases we could use `php://memory` stream https://evertpot.com/222/